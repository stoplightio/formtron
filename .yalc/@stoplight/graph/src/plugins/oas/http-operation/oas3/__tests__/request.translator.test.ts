jest.mock('../helpers/content.translator');
import { mockPassthroughImplementation } from '@stoplight/test-utils';
import { translateMediaTypeObject } from '../helpers/content.translator';
import { translateOas3ToRequest } from '../helpers/request.translator';

describe('translateOas3ToRequest', () => {
  beforeEach(() => {
    mockPassthroughImplementation(translateMediaTypeObject);
  });

  test('given no request body should translate parameters', () => {
    expect(
      translateOas3ToRequest([
        {
          name: 'param-name-1',
          in: 'query',
          content: {
            'content-a': {
              schema: {},
            },
          },
        },
        {
          name: 'param-name-2',
          in: 'query',
          content: {
            'content-b': {
              schema: {},
            },
          },
        },
        {
          name: 'param-name-3',
          in: 'header',
          content: {
            'content-c': {
              schema: {},
            },
          },
        },
      ])
    ).toMatchSnapshot();
  });

  test('give a request body should translate it', () => {
    expect(
      translateOas3ToRequest([], {
        description: 'descr',
        required: true,
        content: {
          'content-a': {
            schema: {},
          },
        },
      })
    ).toMatchSnapshot();
  });
});
