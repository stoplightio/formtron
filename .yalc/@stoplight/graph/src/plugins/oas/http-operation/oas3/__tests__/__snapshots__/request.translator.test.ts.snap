// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`translateOas3ToRequest give a request body should translate it 1`] = `
Object {
  "body": Object {
    "contents": Array [
      "translateMediaTypeObject({\\"schema\\":{}}, content-a, {\\"content-a\\":{\\"schema\\":{}}})",
    ],
    "description": "descr",
    "required": true,
  },
}
`;

exports[`translateOas3ToRequest given no request body should translate parameters 1`] = `
Object {
  "header": Array [
    Object {
      "content": Object {
        "content-c": "translateMediaTypeObject({\\"schema\\":{}}, content-c, {\\"content-c\\":{\\"schema\\":{}}})",
      },
      "name": "param-name-3",
    },
  ],
  "query": Array [
    Object {
      "content": Object {
        "content-a": "translateMediaTypeObject({\\"schema\\":{}}, content-a, {\\"content-a\\":{\\"schema\\":{}}})",
      },
      "name": "param-name-1",
    },
    Object {
      "content": Object {
        "content-b": "translateMediaTypeObject({\\"schema\\":{}}, content-b, {\\"content-b\\":{\\"schema\\":{}}})",
      },
      "name": "param-name-2",
    },
  ],
}
`;
